middleware/block_unsupported_roles.rb

```diff
@@ -0,0 +1,59 @@
+require 'mixins/client_ip'
+
+module CloudFoundry
+  module Middleware
+    class BlockUnsupportedRoles
+      def initialize(app, logger:)
+        @app                   = app
+        @logger                = logger
+      end
+
+      def call(env)
+        current_user = VCAP::CloudController::SecurityContext.current_user
+        if current_user.try(:id) && !globally_authenticated? && config_enabled? && space_supporter_and_only_space_supporter?(current_user)
+          [
+            403,
+            { 'Content-Type' => 'text/html' },
+            'You are not authorized to perform the requested action. See section \'Space Supporter Role in V2\' https://docs.cloudfoundry.org/concepts/roles.html',
+          ]
+        else
+          @app.call(env)
+        end
+      end
+
+      private
+
+      def allowed_path?(path)
+        ['/v2/info', '/'].include?(path)
+      end
+
+      def globally_authenticated?
+        roles.admin? || roles.admin_read_only? || roles.global_auditor?
+      end
+
+      def config_enabled?
+        VCAP::CloudController::Config.config.get(:temporary_enable_space_supporter_role)
+      end
+
+      def space_supporter_and_only_space_supporter?(current_user)
+        user_roles = VCAP::CloudController::SpaceManager.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_MANAGER, :type)).where(user_id: current_user.id).
+                     union(VCAP::CloudController::SpaceDeveloper.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_DEVELOPER, :type)).where(user_id: current_user.id)).
+                     union(VCAP::CloudController::SpaceAuditor.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_AUDITOR, :type)).where(user_id: current_user.id)).
+                     union(VCAP::CloudController::SpaceSupporter.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_SUPPORTER, :type)).where(user_id: current_user.id)).
+                     union(VCAP::CloudController::OrganizationManager.select(Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_MANAGER,
+:type)).where(user_id: current_user.id)).
+                     union(VCAP::CloudController::OrganizationBillingManager.select(Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_BILLING_MANAGER,
+:type)).where(user_id: current_user.id)).
+                     union(VCAP::CloudController::OrganizationAuditor.select(Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_AUDITOR,
+:type)).where(user_id: current_user.id)).
+                     all
```

Comment:
🤯  we don't have any easier ways to do this?  Nothing defined in the user model?

Reply:
We're not seeing anything in the user model.

We could move this into the user model, or at least a function like `unique_roles` or something. Do you think that would be better?

It seems like Cloud Controller prefers skinny models which frankly is an odd design pattern. Happy to not follow it if you think we should move this into the model.

Reply:
I would leave it to your discretion.  I don't totally object to this code living here, because it makes it clearer that it is code we can delete if/when we deprecate & delete v2